pkg/controller/dataimportcron-controller.go

```diff
@@ -1354,3 +1369,42 @@ func GetInitialJobName(cron *cdiv1.DataImportCron) string {
 func getSelector(matchLabels map[string]string) (labels.Selector, error) {
 	return metav1.LabelSelectorAsSelector(&metav1.LabelSelector{MatchLabels: matchLabels})
 }
+
+func inferVolumeModeForSnapshot(ctx context.Context, client client.Client, cron *cdiv1.DataImportCron) (*corev1.PersistentVolumeMode, error) {
+	dv := &cron.Spec.Template
+
+	if explicitVolumeMode := getVolumeModeFromDVSpec(dv); explicitVolumeMode != nil {
+		return explicitVolumeMode, nil
+	}
+
+	inferredPvc := &corev1.PersistentVolumeClaim{
+		Spec: corev1.PersistentVolumeClaimSpec{
+			StorageClassName: cc.GetStorageClassFromDVSpec(dv),
+			VolumeMode:       ptr.To(cdiv1.PersistentVolumeFromStorageProfile),
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					// Doesn't matter
+					corev1.ResourceStorage: resource.MustParse("1Gi"),
+				},
+			},
+		},
+	}
+	if err := dvc.RenderPvc(ctx, client, inferredPvc); err != nil {
```

Comment:
@arnongilboa this will probably not work for backports, wdyt about exporting `pvcFromStorage`?

Reply:
the alternative would be to duplicate it, or,
to not support the upgrade path (only set source vol mode on new snapshots)

Reply:
Exporting `pvcFromStorage` just for backports seems odd. Duplicating `RenderPvc` in the backport seems  cleaner imho.

Reply:
How bad of a backport would it be? I already see some APIs changes that would need to be pulled in

Reply:
afaik the api changes are minimal. Will `pvcFromStorage` export eliminate them?

Reply:
`cdiv1.PersistentVolumeFromStorageProfile` at least, yes

I think that for the backport duplicating some storage profile helpers actually is less aggressive than renderpvc/pvcFromStorage, wdyt?
Would hate touching these funcs on a backport

Reply:
smells like the right direction.

Reply:
> Duplicating `RenderPvc` in the backport seems cleaner imho.

I guess this was also your suggestion, I just misunderstood thinking you wanted to reintroduce it in older releases fully

Reply:
I meant duplicating the helper funcs you need, not specifically this one.